src/bootstrap/src/core/builder/tests.rs

```diff
@@ -1131,25 +1200,42 @@ mod snapshot {
                 .render_steps(), @r"
         [build] rustc 0 <host> -> UnstableBookGen 1 <host>
         [build] rustc 0 <host> -> Rustbook 1 <host>
+        [doc] unstable-book (book) <host>
         [build] llvm <host>
         [build] rustc 0 <host> -> rustc 1 <host>
         [build] rustc 1 <host> -> std 1 <host>
+        [doc] book (book) <host>
+        [doc] book/first-edition (book) <host>
+        [doc] book/second-edition (book) <host>
+        [doc] book/2018-edition (book) <host>
+        [build] rustdoc 1 <host>
+        [doc] rustc 1 <host> -> standalone 2 <host>
```

Comment:
Question: wait, are we building less than we should be? I.e. there are two hosts, $\text{host}\_{1} := \text{"host"}$ and $\text{host}\_{1} := \text{"TEST-TRIPLE-1"}$. Or is this an instance where the "host" is actually the build platform and not actually the host platform? I.e. under the scenario where bootstrap is ran on a different platform than where the stage 1 compiler would be ran (?)

Hm... or taking a step back, what does the "hosts" config actually *mean* in bootstrap?

Reply:
That's a very good question :laughing: When you specify `hosts`, you will essentially override the default host *build* target, see this: https://github.com/rust-lang/rust/blob/d8dca727bac3fbeae14654e05d24cfab88edc753/src/bootstrap/src/core/config/config.rs#L859

In terms of what *hosts* means, the documentation is here: https://github.com/rust-lang/rust/blob/213d946a384b46989f6fd9c8ae9c547b4e354455/src/bootstrap/src/lib.rs#L218. `hosts` determines for what targets we should build host tools, while `targets` determines for what targets we should build the stdlib.

But yeah, it's super confusing.

Reply:
Argh, I think I asked you about this once already, but I keep forgetting the actual distinction.